scipy/special/_logsumexp.py

```diff
@@ -104,7 +104,7 @@ def logsumexp(a, axis=None, b=None, keepdims=False, return_sign=False):
 
     """
     xp = array_namespace(a, b)
-    a, b = xp_broadcast_promote(a, b, ensure_writeable=True, force_floating=True, xp=xp)
+    a, b = xp_broadcast_promote(a, b, copy=True, force_floating=True, xp=xp)
```

Comment:
This parameter was a misnomber from the time when there was only numpy and a copy of a read-only array was always writeable.
The key intent of this flag is to ensure we don't write back to the input, not to ensure that the argument is writeable.
So I've renamed it to match the copy parameter of `astype`.

I'll write it in the docstring once the conflict with https://github.com/scipy/scipy/pull/22695 is resolved.